spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -123,11 +79,19 @@ public void initialize(BootstrapRegistry registry) {
 
 		// promote beans to context
 		registry.addCloseListener(event -> {
-			ZookeeperDiscoveryClient discoveryClient = event.getBootstrapContext().get(ZookeeperDiscoveryClient.class);
+			ZookeeperFunction zookeeperFunction = ((ZookeeperFunction) event.getBootstrapContext().get(ConfigServerInstanceProvider.Function.class));
+			ZookeeperDiscoveryClient discoveryClient = zookeeperFunction.getDiscoveryClient();
 			if (discoveryClient != null) {
 				event.getApplicationContext().getBeanFactory().registerSingleton("zookeeperServiceDiscovery",
 						discoveryClient);
 			}
+
+			CuratorFramework curatorFramework = zookeeperFunction.getCuratorFramework();
```

Comment:
This used to be done by `CuratorFactory.registerCurator` but we don't call that anymore so we need to promote it here